### ISSUE TITLE
Crash report - runtime error: index out of range

### ISSUE THREAD

**Issue by wchargin** [github/hub#1616]
Summary:
I ran `git clone --mirror git://external-repository-url` and then ran
`hub remote add` within the cloned repository, hoping that this would
add my GitHub repo as the remote. It did not.

Possibly noteworthy:
  - I actually don't have a GitHub repo of the exact same name; this
    directory is `blender.git` and I wanted to update `blender`.
  - There is already a remote named `origin`.

The URL that I cloned was: `git://git.blender.org/blender.git`.

Autogenerated report:

Error (runtime.errorString): `runtime error: index out of range`

Stack:

```
goroutine 1 [running]:
panic(0x8ca4a0, 0xc82000e150)
	/home/travis/.gimme/versions/go1.6.3.linux.amd64/src/runtime/panic.go:443 +0x4e9
github.com/github/hub/git.WorkdirName(0x0, 0x0, 0x0, 0x0)
	/tmp/go/src/github.com/github/hub/git/git.go:70 +0xf7
github.com/github/hub/commands.transformRemoteArgs(0xc820098e60)
	/tmp/go/src/github.com/github/hub/commands/remote.go:66 +0x710
github.com/github/hub/commands.remote(0xbf2da0, 0xc820098e60)
	/tmp/go/src/github.com/github/hub/commands/remote.go:42 +0x89
github.com/github/hub/commands.(*Command).Call(0xbf2da0, 0xc820098e60, 0x0, 0x0)
	/tmp/go/src/github.com/github/hub/commands/commands.go:47 +0x188
github.com/github/hub/commands.(*Runner).Call(0xc820073450, 0xbf2da0, 0xc820098e60, 0x0, 0x0, 0x0, 0x0)
	/tmp/go/src/github.com/github/hub/commands/runner.go:108 +0x5e
github.com/github/hub/commands.(*Runner).Execute(0xc820073450, 0x0, 0x0, 0x0, 0x0)
	/tmp/go/src/github.com/github/hub/commands/runner.go:96 +0x1df
main.main()
	/home/travis/gopath/src/github.com/github/hub/main.go:14 +0x50

```

Runtime:

```
GOOS: linux
GOARCH: amd64
```

Version:

```
git version 2.14.2
hub version 2.2.9
```

**Comment by wchargin** [comment]
Looks like this might have been fixed in 01c3099abe56a8dcca41e285c11c0f6f860cc338?

**Comment by mislav** [comment]
Yes, since `--mirror` implies `--bare`, this was covered by that fix. Thanks for reporting, though!
